dask_expr/groupby.py

```diff
@@ -297,6 +307,16 @@ def combine_kwargs(self):
     def _divisions(self):
         return (None, None)
 
+    def _simplify_up(self, parent):
+        if isinstance(parent, Projection):
+            columns = sorted(set(parent.columns + self.by))
+            if columns == self.frame.columns:
+                return
+            return type(parent)(
+                type(self)(self.frame[columns], *self.operands[1:]),
+                *parent.operands[1:],
+            )
```

Comment:
Logic looks fine here. Just leaving a note that this could also be `return type(self)(self.frame[columns], *self.operands[1:])[parent.operand("columns")]` (though I'm not sure that is any more "readable" - so, I don't have much of a personal preference)